tests/ops/functions/test_map_wires.py

```diff
@@ -0,0 +1,211 @@
+        assert s_op.data == mapped_op.data
+        assert s_op.wires == mapped_op.wires
+        assert s_op.arithmetic_depth == mapped_op.arithmetic_depth
+
+    def test_execute_mapped_tape(self):
+        """Test the execution of a mapped tape."""
+        dev = qml.device("default.qubit", wires=5)
+        tape = QuantumTape()
+        with tape:
+            build_op()
+            qml.expval(op=qml.PauliZ(1))
+
+        # TODO: Use qml.equal when supported
+
+        m_tape = qml.map_wires(tape, wire_map=wire_map)
+        m_op = m_tape.operations[0]
+        m_obs = m_tape.observables[0]
+        assert isinstance(m_op, qml.ops.Prod)
+        assert m_op.data == mapped_op.data
+        assert m_op.wires == mapped_op.wires
+        assert m_op.arithmetic_depth == mapped_op.arithmetic_depth
+        assert m_obs.wires == Wires(wire_map[1])
+        assert qml.math.allclose(dev.execute(tape), dev.execute(m_tape))
+
+
+class TestMapWiresQNodes:
+    """Tests for the qml.map_wires method used with qnodes."""
+
+    def test_map_wires_qnode(self):
+        """Test the map_wires method with a qnode."""
+        dev = qml.device("default.qubit", wires=5)
+
+        @qml.qnode(dev)
+        def qnode():
+            build_op()
+            return qml.expval(op=build_op())
+
+        # TODO: Use qml.equal when supported
+
+        m_qnode = qml.map_wires(qnode, wire_map=wire_map)
+        assert m_qnode() == qnode()
+        assert len(m_qnode.tape) == 2
+        m_op = m_qnode.tape.operations[0]
+        m_obs = m_qnode.tape.observables[0]
+        assert isinstance(m_op, qml.ops.Prod)
+        assert m_op.data == mapped_op.data
+        assert m_op.wires == mapped_op.wires
+        assert m_op.arithmetic_depth == mapped_op.arithmetic_depth
+        assert isinstance(m_obs, qml.ops.Prod)
+        assert m_obs.data == mapped_op.data
+        assert m_obs.wires == mapped_op.wires
+        assert m_obs.arithmetic_depth == mapped_op.arithmetic_depth
+
+
+class TestMapWiresCallables:
+    """Tests for the qml.map_wires method used with callables."""
+
+    def test_map_wires_qfunc(self):
+        """Test the map_wires method with a qfunc."""
+        dev = qml.device("default.qubit", wires=5)
+
+        def qfunc():
+            build_op()
+            return qml.probs(wires=0)
+
+        m_qfunc = qml.map_wires(qfunc, wire_map=wire_map)
+
+        qnode = qml.QNode(qfunc, dev)
+        m_qnode = qml.QNode(m_qfunc, dev)
+
+        # TODO: Use qml.equal when supported
+
+        assert qml.math.allclose(m_qnode(), qnode())
+        assert len(m_qnode.tape) == 2
+        m_op = m_qnode.tape.operations[0]
+        assert isinstance(m_op, qml.ops.Prod)
+        assert m_op.data == mapped_op.data
+        assert m_op.wires == mapped_op.wires
+        assert m_op.arithmetic_depth == mapped_op.arithmetic_depth
+        assert m_qnode.tape.observables[0].wires == Wires(wire_map[0])
+
+    @pytest.mark.jax
+    def test_jitting_simplified_qfunc(self):
```

Comment:
Worth testing with other interfaces as well?

Reply:
This is testing that the circuit can be compiled using jax.

I don't think we should add more tests with other interfaces given that wires is independent of the interface